src/Shared/Recruit.Vacancies.Client/Infrastructure/SequenceStore/MongoSequenceStore.cs

```diff
@@ -0,0 +1,32 @@
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Services;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Mongo;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.SequenceStore
+{
+    internal sealed class MongoSequenceStore : MongoDbCollectionBase, IGenerateVacancyNumbers
+    {
+        private const string Database = "recruit";
+        private const string Collection = "sequences";
+        private const string VacancyNumberSequenceName = "vacancy-number";
+
+        public MongoSequenceStore(IOptions<MongoDbConnectionDetails> details)
+            : base(Database, Collection, details)
+        {
+        }
+
+        public async Task<long> GenerateAsync()
+        {
+            var collection = GetCollection<Sequence>();
+            var filter = Builders<Sequence>.Filter.Eq(x => x.Name, VacancyNumberSequenceName);
```

Comment:
Could the Id just be the name? Similar to what has been done in some of the queryviews such as for Dashboard and EditVacancyInfo